File: leetcode/0191. Number of 1 Bits/Solution.swift
// https://leetcode.com/problems/number-of-1-bits/

func hammingWeight(_ n: Int) -> Int {
  var c = 0, m = n
  while m > 0 {
    c += m & 1
    m >>= 1
  }

  return c
}

assert(hammingWeight(0b00000000000000000000000000001011) == 3)
assert(hammingWeight(0b00000000000000000000000010000000) == 1)
assert(hammingWeight(0b11111111111111111111111111111101) == 31)

File: leetcode/0191. Number of 1 Bits/Solution.py
# https://leetcode.com/problems/number-of-1-bits/

class Solution:
    def hammingWeight(self, n: int) -> int:
        w = 0
        while n:
            w += n & 1
            n >>= 1
        
        return w

s = Solution()
assert s.hammingWeight(0b00000000000000000000000000001011) == 3
assert s.hammingWeight(0b00000000000000000000000010000000) == 1
assert s.hammingWeight(0b11111111111111111111111111111101) == 31